### PR TITLE
Fix #16

### LINKED ISSUE #16: Provide start values for variables that requires it
According to the spec, some variables with a given causality/variability must provide a start value. This is currently not the case. At least not automatically.

### PR DESCRIPTION
**Pull request by markaren**
@fcollonval Care to have a look? I did not figure out how to set the instances `__attrd` from outside the class, so I made a setter. Also it should be tested, but I dunno how to access the start xml element using pyfmi.

Closes #16

### CODE REVIEW

File: pythonfmu/variables.py [hunk 1]
```diff
@@ -68,6 +68,9 @@ def __init__(self, name: str, start: Optional[Any] = None, **kwargs):
         super().__init__(name, **kwargs)
         self.__attrs = {"start": start}
 
+    def __set_start__(self, value: float):
```
Comment (fcollonval) [inline pythonfmu/variables.py]:
```suggestion
    @start.setter
    def start(self, value: float):
```

in python, that is the syntax to define the setter of a property (see [documentation](https://docs.python.org/3/library/functions.html#property) for more info).

Then you can set the variable using:
```python
var.start = value
```

File: pythonfmu/fmi2slave.py [hunk 2]
```diff
@@ -94,6 +96,33 @@ def to_xml(self, model_options: Dict[str, str] = dict()) -> Element:
 
         return root
 
+    @staticmethod
+    def __requires_start__(v: ScalarVariable) -> bool:
```
[the remark this answers is not in the document]
Comment (fcollonval) [reply pythonfmu/fmi2slave.py]: This will make more sense as `ScalarVariable` method.

File: pythonfmu/fmi2slave.py [hunk 3]
```diff
@@ -82,6 +82,8 @@ def to_xml(self, model_options: Dict[str, str] = dict()) -> Element:
 
         variables = SubElement(root, 'ModelVariables')
         for v in self.vars.values():
+            if self.__requires_start__(v):
```
Comment (fcollonval) [inline pythonfmu/fmi2slave.py]:
```suggestion
            if Fmi2Slave.__requires_start__(v):
```

**Comment by markaren** [comment]
Regarding the test, I got an error with your code and the code itself is completely Greek to me. 
However, I made this work:

```python
xml = slave.to_xml()
    var_node = xml.find(".//ScalarVariable[@name='var']")
    assert var_node is not None

    if ScalarVariable.requires_start(var_obj):
        assert var_obj.start is value
    else:
        assert var_obj.start is None
```

Looks correct no?

File: pythonfmu/fmi2slave.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (markaren) [reply pythonfmu/fmi2slave.py]: I moved it to ScalarVariable. Btw, do you know why typehinting `ScalarVariable` does not work inside the `ScalarVariable` itself?

Also probably I should make a PR renaming the getters and setters which do the special__XXX__ thingy. __XXX would suffice.

[the remark this answers is not in the document]
Comment (fcollonval) [reply pythonfmu/fmi2slave.py]: > do you know why typehinting ScalarVariable does not work inside the ScalarVariable itself?

This is related to Python type hint being annotation. And Python cannot evaluate type without evaluating the all associated code. The way around that is to define the type as string.

```python
def requires_start(v: "ScalarVariable") -> bool:
```

Regarding the getter/setter, I'm working on a PR right now.

File: pythonfmu/tests/test_variables.py [hunk 4]
```diff
@@ -58,6 +59,42 @@ def test_ScalarVariable_to_xml(causality, initial, variability, name, descriptio
...
+            return True
+
+    slave = PySlave()
+
+    xml = slave.to_xml()
+    var_node = xml.find(".//ScalarVariable[@name='var']")
+    assert var_node is not None
+
+    if ScalarVariable.requires_start(var_obj):
+        assert var_obj.start is value
```
Comment (fcollonval) [inline pythonfmu/tests/test_variables.py]:
```suggestion
        assert var_obj.start == value
```

Another python feature, `is` compares object type. So equality `==` test should be used here.

**Pull request merged**